### PR TITLE
feat: moved get-involved and guides content

### PR DESCRIPTION
**Pull request by ovflowd**
## Description

This Pull Request is another push towards the Navigation that will exist on the Node.js Website Redesign.

This Pull Request moves all `/get-involved` pages unto `/about/get-involved` and it moves the `/docs/guides` into a dedicated `/guides` top-level navigation.

### Other Related Changes

- `/:locale/docs` will now redirect to `https://nodejs.org/docs/latest/api/`
- `/:locale/docs/guides` will now redirect to `/:locale/guides`
  - All subsequent pages will also be appropriately redirected to their 1:1 match
- The `/:locale/docs/es6` will redirect to the ES6 learn page of the same content
- The Top Navigation now contains "Guides"
- The Top Navigation "Docs" will now point to `https://nodejs.org/docs/latest/api/`
- We added a Blockquote about the "deprecation" of the Guides and that they will eventually be replaced by respective Learn pages
- Updated Content (Markdown) with proper new URLs and redirects

## Validation

- /en/docs -> https://nodejs.org/docs/latest/api/
- /en/docs/guides/ && /en/docs/guides/*** to -> /en/guides && /en/guides/***
- /en/docs/es6 -> /en/learn/getting-started/ecmascript-2015-es6-and-beyond
- Proper new Top Navigations
- No Broken Links on Guides Pages
- Sidebars still displaying properly

### CODE REVIEW

**Comment by ovflowd** [comment]
cc @nodejs/tsc as this is a relatively big "content-drift" on the Website. Note that this change does **not** remove content; Just reallocate them to different places to properly start to match the Website Redesign.

**Comment by ovflowd** [comment]
**Note:** All existing links will still work and properly redirect. This PR should **not** break existing links and user experience.

**Review by mcollina — APPROVED** [review]
lgtm

**Review by ShogunPanda — APPROVED** [review]
LGTM!

**Review by marco-ippolito — APPROVED** [review]

**Review by himself65 — APPROVED** [review]

**Review by AugustinMauroy — APPROVED** [review]

File: redirects.json [hunk 1]
```diff
@@ -173,6 +171,35 @@
...
+      "source": "/:locale/get-involved/:path*",
+      "destination": "/:locale/about/get-involved/:path*"
+    },
+    {
+      "source": "/:locale/learn",
+      "destination": "/:locale/learn/getting-started/introduction-to-nodejs"
     }
-  ]
+  ],
+  "internal": []
```
Comment (bmuenzenmeyer) [inline redirects.json:204]: with internal now empty, how relevant is this code within https://github.com/nodejs/nodejs.org/blob/8ef12a4728c4553ad7a1bbfc50fad88d6f2f107b/next.rewrites.mjs#L41

**Review by bmuenzenmeyer — APPROVED** [review]
one question about the `next.rewrites.js` logic - functionally this looks good

File: redirects.json (hunk 1 above)
Comment (ovflowd) [reply redirects.json:204]: So, the `internal` redirects are actually rewrites. I.e. you are on route A but render content of route B;

And so far, for all use cases we have, we don't want these to happen, at least for what I can tell. We should still keep this logic if we eventually want those to happen.

With "rewrites" /en/get-involved -> /en/about/get-involved "rewrite" would keep the /en/get-involved URL but render /en/about/get-involved; But we actually want Route A to be redirected to Route B

**Review by aymen94 — APPROVED** [review]

**Review by mhdawson — APPROVED** [review]
LGTM

File: components/SideNavigation.tsx [hunk 2]
```diff
@@ -6,37 +6,31 @@ import { useSiteNavigation } from '@/hooks/server';
...
 
-  const [[, sideNavigationItems]] = getSideNavigation([navigationKey], context);
+  const sideNavigation = getSideNavigation(navigationKeys, context);
+
+  const mapItems = (items: ReturnType<typeof getSideNavigation>) => {
+    return items.map(([, { link, label, items }]) => (
+      <li key={link}>
+        {link ? <ActiveLink href={link}>{label}</ActiveLink> : label}
+
+        {items && items.length > 0 && <ul>{mapItems(items)}</ul>}
```
Comment (JakobJingleheimer) [inline components/SideNavigation.tsx:26]: nit:

```suggestion
        {!!items?.length && <ul>{mapItems(items)}</ul>}
```

Comment (ovflowd) [reply components/SideNavigation.tsx:26]: I feel that the current one might be easier to read/understand for newcomers, no?

**Review by RafaelGSS — APPROVED** [review]
LGTM.

**Review by UlisesGascon — APPROVED** [review]
LGTM!

File: components/SideNavigation.tsx (hunk 2 above)
Comment (ovflowd) [reply components/SideNavigation.tsx:26]: @JakobJingleheimer can I resolve this thread?

**Pull request merged**